src/main/java/frc/robot/subsystems/hood/Hood.java

```diff
@@ -0,0 +1,68 @@
+package frc.robot.subsystems.hood;
+
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import static frc.robot.Ports.Hood.SOLENOID;
+
+public class Hood extends SubsystemBase {
+    private static Hood INSTANCE;
+    private final Solenoid angleChanger = new Solenoid(PneumaticsModuleType.CTREPCM, SOLENOID);
+
+    private Hood() {
+    }
+
+    public enum Mode {
+        ShortDistance(true),
+        LongDistance(false);
+        public final boolean value;
+
+        Mode(boolean value) {
+            this.value = value;
+        }
+    }
+
+    /**
+     * Gets the instance of the hood subsystem.
+     *
+     * @return the hood subsystem.
+     */
+    public static Hood getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Hood();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * Opens the solenoid.
+     * This function simply toggles the mode of the
+     * solenoid only if it's currently closed.
+     */
+    public void open() {
+        angleChanger.set(false);
+    }
+
+    /**
+     * Closes the solenoid.
+     * This function simply toggles the mode of the
+     * solenoid only if it's currently open.
+     */
+    public void close() {
+        angleChanger.set(true);
+    }
+
+    public void setSolenoid(Mode mode){
+        angleChanger.set(mode.value);
+    }
+
+    /**
+     * Returns whether the solenoid is active.
+     *
+     * @return a boolean representing the mode of the solenoid.
```

Comment:
```suggestion
     * @return whether the solenoid is active.
```
However, "solenoid" shouldn't be here, don't expose hardware components out.